webrecorder/browsertrix
# Standardize archived item filtering, sorting and labels

Resolves https://github.com/webrecorder/browsertrix-cloud/issues/1025



### Changes

- Renames list view to "All Archived Items"
- Refactors fetches to use single `all-crawls` endpoints
- Removes search by config ID for more search parity with uploads
- Adds sort by size
- Refactors property and method names to replace `crawl*`
- Replaces remaining references to "crawl" in copy with "item"'

The view/status filter needs some re-thinking, since uploads cannot be partially complete or timed out. Maybe the "All"/"Crawls"/"Uploads" can be moved into the filter control area, and choosing one of the types reveals more filters in a more predictable way.

### Manual testing

1. Log in and click "All Archived Items"
2. Sort by Size. Verify both crawls and uploads are sorted
3. Type some text into search bar. Verify both crawls and uploads are shown in results
4. Click a search result. Verify list is filtered as expected
5. Repeat 2-4 for Crawls and Uploads to test against regressions

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| All Archived Items | <img width="1138" alt="Screen Shot 2023-08-09 at 2 02 49 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/6758804/de160047-c9f2-4234-8da9-2279890f7e9c"> |
